### DISCUSSION TITLE
selecting/uploading pdf file

### DISCUSSION THREAD

**Discussion by linpi1** [joomengine/Joomla-Component-Builder#803] [Extention Issues]
in a repeatable field, I'm using the field media to be able to select/upload pdf files. It works for images, but pdf are not even seen. Where can I tweak some code so pdf are allowed ? thanks for a little hint. 
In the website, I'm using the editor JCE, and JCE definitively accepts pdfs.

**Comment by Llewellynvdm** [comment]
Sorry @linpi1 I can't say that I have manage that before in any shape or form. Yes I have worked with PDF's and can display them but this is not easy task from my point of view, maybe I just missed something simple as you pointed to JCE as being able to accept PDF's... hmm I am not even sure what you mean by that.

### How to resolve this?

Well JCB just build what Joomla needs to again build your views... forms and stuff. So in the end it is Joomla that display the images... not JCB. Joomla builds the repeatable fields and the media fields and it does it from the XML JCB builds, yet this XML is the Joomla way of doing this... So you could ask this question on the [Joomla forum](https://forum.joomla.org/), or on the [Joomla stackoverflow](https://joomla.stackexchange.com/) to see if anyone there is able to help you out.

Should you get and answer somewhere, please share the link here :+1:

**Reply by Llewellynvdm** [reply]
Allowing the upload of PDF's this is done in the media manger itself if I remember correctly... so if that is all your really looking for is a way to upload PDF's then look around the media manger global options.

But to display this PDF like the image... that is what my first reply was all about.

**Reply by linpi1** [reply]
yes you are right, there is possible to add pdf in the allowed extensions, and I did this already. In Media, I can see or upload pdfs.

**Comment by TLWebdesign** [comment]
Hi, the media type field only display images by default. It's just how that field is programmed. Do you need upload functionality or just being able to select pdf's? If only selection is needed you could use filelist field type.

If you need media manager with pdf capabilities. You could check this tutorial out on how to create a plugin that does the necessary changes to the media manager to allow pdf's to be visible in the modal of the media manager: https://www.richardbutterworth.co.uk/blog/19-the-joomla-media-form-field

**Reply by TLWebdesign** [reply]
Hi @linpi1,

got a working plugin for JCB. I added the plugin to a JCB package including a basic component and the plugin.

You can import this package in your JCB installation and check it out and assign it to your own component.
Make sure to assign the "overrides" custom folder to your "media" folder in your component's setting. By importing this package i'm almost sure it also uploads the "overrides" folder to your "custom" folder in the  JCB administrator folder.

https://github.com/TLWebdesign/MediaFieldShowDocs-Plugin-JCB-Package

**Reply by linpi1** [reply]
Hi Tom, 
sorry that  couldn't get back earlier. Thanks a lot for you work, I'm very grateful. I installed this plugin & compiled. I can know the  component "Media Field Show Docs", the admin view "Media test", the plugin "MediaFieldShowDocs" & the override custom folder in the settings of the component.  But I when I try a "new mediatest", still I can not upload a pdf file nor see the existing ones (in the Media/com_media, I can/upload pdfs... )
It's really strange.

**Reply by TLWebdesign** [reply]
Did you check if the plugin is activated? you need to activate it after installation.

**Reply by linpi1** [reply]
no, this is problem. There is no plugin MediaFieldShowDocs in the joomla plugins, not even deactivated. ok, I learning know how plugins in JBC works, and will get back to you soon.

**Reply by linpi1** [reply]
yes great, it works! at least when the editor in the global config is set to TinyMCE. With JCE, doesn't seem to work, yet...

**Reply by TLWebdesign** [reply]
I wouldn't know how the editor has influence on this since the field is not an "editor" field. Will have a look
